vpr/src/route/route_timing.cpp

```diff
@@ -579,11 +588,17 @@ bool try_timing_driven_route_tmpl(const t_router_opts& router_opts,
             pres_fac = update_pres_fac(router_opts.initial_pres_fac);
         }
 
+        if (budgeting_inf.if_set()) {
+            // If total negative slack is greater than or equal to 0, or RCV is finished, allow router to complete
+            hold_legal = timing_info->hold_total_negative_slack() >= 0 || rcv_finished_count == 0;
```

Comment:
If this variable doesn't really mean hold is legal (it seems to mean RCV should stop trying) then you should give it a different name.